azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/containerapp/ContainerAppDraft.java

```diff
@@ -275,6 +278,73 @@ private static void tarSourceIfNeeded(final BuildImageConfig buildConfig) {
         }
     }
 
+    private String buildThroughACR(final ImageConfig imageConfig, final BuildImageConfig buildConfig) {
+        Map<String, OverridingArgument> overridingArguments = Optional.ofNullable(imageConfig.getBuildImageConfig())
+            .map(BuildImageConfig::getSourceBuildEnv)
+            .orElse(Collections.emptyMap())
+            .entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, e -> new OverridingArgument(e.getValue(), false)));
+        final ContainerRegistry registry = getOrCreateRegistry(imageConfig);
+        tarSourceIfNeeded(buildConfig);
+        final RegistryTaskRun run = registry.buildImage(imageConfig.getAcrImageNameWithTag(), buildConfig.getSource(), "./Dockerfile", overridingArguments);
+        if (Objects.isNull(run)) {
+            throw new AzureToolkitRuntimeException("ACR is not ready, Failed to build image through ACR.");
```

Comment:
Can we get more error information out of the failure? If we just output this, it is difficult to understand what went wrong in the background, and what the customer can do to recover.

Reply:
Here is the logic intellij plugin throw before and I just extract it for reuse. If the run object is null, I think it just means the ACR is not ready. But if the acr task build fails, we can output more info about the build log.